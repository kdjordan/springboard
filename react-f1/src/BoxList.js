import React from 'react'
import Box from './Box'
import './BoxList.css'

function BoxList({data, removeBox}) {
    console.log('gettint ', data)
    const deleteBox = (e, boxId) => {
        removeBox(() => e, boxId)
    }

 return (
    <div className='BoxList'>
        <h2>Currrent Boxes</h2>
        <div className="BoxList-container">
            { data.map(box =>  {
                return (
                    <Box id={box.id} height={box.height} width={box.width} color={box.color} deleteBox={deleteBox}/>
                )
        })}
        </div>
    </div>
 )

}

export default BoxList