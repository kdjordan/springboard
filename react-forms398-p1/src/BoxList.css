.BoxList  h2{
    margin: 2vh 0;
}

.BoxList-container {
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    gap: 1rem;
    flex-wrap: wrap;
    max-width: 
}