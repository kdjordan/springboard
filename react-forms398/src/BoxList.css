.BoxList {
    margin-top: 2vh;
}