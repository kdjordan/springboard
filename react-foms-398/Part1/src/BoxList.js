import React, {useState} from 'react'
import Box from './Box'
import NewBoxForm from './NewBoxForm'

import './BoxList.css'
import { v4 as uuidv4 } from 'uuid';

function BoxList() {
    const [allBoxes, setAllBoxes] = useState([])

    const addBox = (e, { height, width, color }) => {
        //set default color if nothing is passed
        if(!color) {
        color = '#000000'
        }
        //enforce max height and width
        if(Number(height) > 200) {
        height = '200'
        }
        if(Number(width) > 200) {
        height = '200'
        }
        //create newBox object
        //add in random id
        let newBox = {
            id: uuidv4(),
            height: `${Number(height)}px`,
            width: `${Number(width)}px`,
            color
        }
        //update state with new Box
        setAllBoxes(boxes => [...boxes, newBox])
    }
  //use filter to revome box by ID
  const removeBox = (e, boxId) => {
    setAllBoxes(boxes => {
      return boxes.filter(b => {
        return b.id !== boxId
      })
    })
    }

 return (
    <div className='BoxesList'>
        <NewBoxForm addBox={addBox} />
        <h2>Currrent Boxes</h2>
        <div className="BoxList-container" data-testid='boxlist-container'>
            { allBoxes.map(box =>  {
                return (
                    <Box 
                    key={box.id}
                    id={box.id} 
                    height={box.height} 
                    width={box.width} 
                    color={box.color} 
                    handleDelete={removeBox}
                    />
                )
            })}
        </div>
    </div>
 )

}

export default BoxList