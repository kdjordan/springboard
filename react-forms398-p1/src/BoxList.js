import React from 'react'
import Box from './Box'
import './BoxList.css'

function BoxList(props) {
    console.log('gettint ', props)
    const deleteBox = (e, boxId) => {
        props.removeBox(() => e, boxId)
    }

 return (
    <div className='BoxList'>
        <h2>Currrent Boxes</h2>
        <div className="BoxList-container">
            { props.data.map(box =>  {
                return (
                    <Box id={box.id} height={box.height} width={box.width} color={box.color} deleteBox={deleteBox}/>
                )
        })}
        </div>
    </div>
 )

}

export default BoxList