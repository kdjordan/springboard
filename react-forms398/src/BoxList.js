import React, { useState } from 'react'
import Box from './Box'
import './BoxList.css'
import { v4 as uuidv4 } from 'uuid';

function BoxList({data}) {
    console.log(data)
    const [boxes, setBoxes] = useState()
    const deleteBox = (e, id) => {
        console.log('delete Box', id)
    }

 return (
    <div className='BoxList'>
        <h2>Currrent Boxes</h2>
        <div className="BoxList-container">
            { data.map(box =>  {
                return (
                    <Box id={uuidv4()} height={box.height} width={box.width} color={box.color} deleteBox={deleteBox}/>
                )
        })}
        </div>
    </div>
 )

}

export default BoxList