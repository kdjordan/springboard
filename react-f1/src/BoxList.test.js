import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function() {
  render(<BoxList data={[{key:'88', id:'test', height: '100px', width:'100px', color:'#000000'}]}/>);
}); 

it("matches snapshot", function() {
  const {asFragment} = render(<BoxList data={[{key:'88', id:'test', height: '100px', width:'100px', color:'#000000'}]}/>);
  expect(asFragment()).toMatchSnapshot();
});

it("matches diff snapshot", function() {
  const {asFragment} = render(<BoxList data={[{key:'88', id:'test', height: '100px', width:'100px', color:'#ffffff'}]}/>);
  expect(asFragment()).toMatchSnapshot();
});

it(('should call passed function on form submit'), function() {
  const box = {id: 'ee4e', color:'#000000', height:'100px', width:'100px'}
  const deleteBox = jest.fn();
  render(<BoxList />)
  const boxBtn = screen.
  const boxWrapper = screen.getByTestId('Current-boxes')
  fireEvent.submit(form)
  expect(deleteBox).toHaveBeenCalled();
})