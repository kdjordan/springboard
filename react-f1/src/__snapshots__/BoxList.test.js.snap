// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches diff snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h2>
      Currrent Boxes
    </h2>
    <div
      class="BoxList-container"
    >
      <div
        class="BoxWrapper"
      >
        <div
          class="Box"
          style="height: 100px; width: 100px; background-color: rgb(255, 255, 255);"
        />
        <button
          class="BoxButton"
        >
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h2>
      Currrent Boxes
    </h2>
    <div
      class="BoxList-container"
    >
      <div
        class="BoxWrapper"
      >
        <div
          class="Box"
          style="height: 100px; width: 100px; background-color: rgb(0, 0, 0);"
        />
        <button
          class="BoxButton"
        >
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
